src/api_client.rs

```diff
@@ -395,8 +395,18 @@ impl Client {
         unimplemented!("")
     }
 
-    pub async fn get_health() -> Result<HealthStatus, Error> {
-        unimplemented!("")
+    pub async fn get_health(&self) -> Result<HealthStatus, Error> {
+        let path = "eth/v1/node/health";
+        let target = self.endpoint.join(path)?;
+        let request = self.http.get(target);
+        let response = request.send().await?;
+        let result = match response.status() {
+            http::StatusCode::OK => HealthStatus::Ready,
+            http::StatusCode::PARTIAL_CONTENT => HealthStatus::Syncing,
+            http::StatusCode::SERVICE_UNAVAILABLE => HealthStatus::NotInitialized,
```

Comment:
minor stylistic preference to `use http::StatusCode;` at the top of the file and then just match on `StatusCode::X`

Reply:
ah yea ok. done.